### ISSUE TITLE
bug: appending to list in document tags

### ISSUE THREAD

**Issue by slettner** [jina-ai/jina#3209]
**Describe the bug**
When using a python list as a value for the `Document` tags appending to the list raise `AttributeError: insert`.

Reproduce
```
from jina import Document

doc = Document()
doc.tags['key'] = []
doc.tags['key'].append(1)
```

Traceback:
```
Traceback (most recent call last):
  File "test.py", line 5, in <module>
    doc.tags['key'].append(1)
  File "/usr/local/Cellar/python@3.8/3.8.10/Frameworks/Python.framework/Versions/3.8/lib/python3.8/_collections_abc.py", line 962, in append
    self.insert(len(self), value)
  File "/Users/sebastianlettner/Desktop/tmp/.venv/lib/python3.8/site-packages/jina/types/list.py", line 31, in insert
    self._pb_body.insert(index, object)
AttributeError: insert
```

**Describe how you solve it**
Get the list and convert it, append and re-assign the full list.
````
from jina import Document

doc = Document()
doc.tags['key'] = [] 
l = list(doc.tags.get('key'))
l.append(1)
doc.tags['key'] = l
````

---

**Environment**
```
- jina 2.0.18
- jina-proto 0.0.85
- jina-vcs-tag (unset)
- libzmq 4.3.4
- pyzmq 1.21.2
- protobuf 3.17.3
- proto-backend cpp
- grpcio 1.39.0
- pyyaml 5.4.1
- python 3.8.10
- platform Darwin
- platform-release 19.6.0
- platform-version Darwin Kernel Version 19.6.0: Thu Oct 29 22:56:45 PDT 2020;
  root:xnu-6153.141.2.2~1/RELEASE_X86_64
- architecture x86_64
- processor i386
- uid 190070690681122
- session-id 2d9e1a1a-000d-11ec-a253-acde48001122
- uptime 2021-08-18T12:15:13.759193
- ci-vendor (unset)
* JINA_ARRAY_QUANT (unset)
* JINA_CONTROL_PORT (unset)
* JINA_DEFAULT_HOST (unset)
* JINA_DISABLE_UVLOOP (unset)
* JINA_FULL_CLI (unset)
* JINA_HUBBLE_REGISTRY (unset)
* JINA_HUB_CACHE_DIR (unset)
* JINA_HUB_ROOT (unset)
* JINA_LOG_CONFIG (unset)
* JINA_LOG_ID (unset)
* JINA_LOG_LEVEL (unset)
* JINA_LOG_NO_COLOR (unset)
* JINA_LOG_WORKSPACE (unset)
* JINA_OPTIMIZER_TRIAL_WORKSPACE(unset)
* JINA_POD_NAME (unset)
* JINA_RANDOM_PORT_MAX (unset)
* JINA_RANDOM_PORT_MIN (unset)
* JINA_VCS_VERSION (unset)
* JINA_MP_START_METHOD (unset)
```

**Comment by JoanFM** [comment]
Will try to take a look. ListView and StructView are tricky

**Comment by JoanFM** [comment]
Solved by #3212
